services/apps/alcs/src/utils/search/search-intersection.ts

```diff
@@ -0,0 +1,18 @@
+import { intersectSets } from '../set-helper';
+
+export const processSearchPromises = async (
+  promises: Promise<{ fileNumber: string }[]>[],
+) => {
+  const queryResults = await Promise.all(promises);
+
+  const allIds: Set<string>[] = [];
+  for (const result of queryResults) {
+    const fileNumbers = new Set<string>();
+    result.forEach((currentValue) => {
+      fileNumbers.add(currentValue.fileNumber);
+    });
+    allIds.push(fileNumbers);
```

Comment:
How come `fileNumbers` uses `.add` and `allIds` uses `.push` when they're both `Set`, is it just because we already initiated an empty set for `allIds`?

Reply:
`allIds` is an Array of Sets which is why it uses `push`

Reply:
Ooh I see it now, thanks!